osu.Game.Rulesets.Mania/Skinning/ManiaLegacySkinTransformer.cs

```diff
@@ -88,10 +90,11 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                     switch (maniaComponent.Component)
                     {
                         case ManiaSkinComponents.ColumnBackground:
-                            return new LegacyColumnBackground(maniaComponent.TargetColumn == beatmap.TotalColumns - 1);
+                            return new LegacyColumnBackground(maniaComponent.TargetColumn == beatmap.TotalColumns - 1, stageBackground);
```

Comment:
i guess this code is the dodgiest of it all, storing a reference based on access order. i can imagine this potentially falling over in the future if this was to change for whatever reason. i guess we can't use DI since the columns aren't contained within the background..

Reply:
Initially I had DI of `Stage` and accessing via `(Stage.Background.Drawable as LegacyStageBackground)?.AddColumnBackground()`, but I didn't feel happy exposing the background like that.

Yeah, it's a bit unfortunate that the columns aren't contained within the background. I had another idea in mind that is to extract the backgrounds from the columns, but I'm not sure how I feel about that path either... I'd like columns to be constructible by themselves for e.g. tests, so that'd have to change.

Reply:
I think I'm okay with it this way if you make sure it fails at least one test on null (ie. remove the `CanBeNull` allowance and cause an exception if ever called out of order). Else we may regress this without even knowing, since it's a legacy-only pathway.

Reply:
I don't understand what you mean by making sure it fails at least one test - it does fail every test in which a stage isn't created (e.g. `TestSceneColumnBackground`).

Reply:
I'll look into rethinking how column backgrounds are added for legacy skins to instead add them directly into `LegacyStageBackground` rather than proxying them in.

Reply:
Done, changes kinda blew up but see if you like it more.